### PR TITLE
Some results about path-cosplit maps

### PR DESCRIPTION
**Pull request by fredrik-bakke**
- When the domain is $n+r+2$-truncated the type of $n$-path-cosplittings is $r$-truncated
- When the domain is $n+1$-truncated $n$-path-cosplittings are unique
- When the domain is $n$-truncated there are unique $n$-path-cosplittings
- path-cosplittings are closed under path-cosplit morphisms of arrows
- path-cosplittings are closed under equivalences of arrows
- path-cosplittings are closed under homotopies
- 3-for-2 of path-cosplittings
- products of path-cosplittings
- total space path-cosplittings
- dependent products of path-coplittings
- coproducts of path-cosplittings

### CODE REVIEW

File: src/foundation/path-algebra.lagda.md [hunk 1]
```diff
@@ -128,6 +128,50 @@ module _
...
+
+  left-right-inv : (p : x ＝ y) (q : x ＝ y) → p ∙ (inv p ∙ q) ＝ q
+  left-right-inv p q = inv (assoc p (inv p) q) ∙ ap (_∙ q) (right-inv p)
+
+  right-left-inv : (p : x ＝ y) (q : z ＝ y) → (p ∙ inv q) ∙ q ＝ p
+  right-left-inv p q = assoc p (inv q) q ∙ ap (p ∙_) (left-inv q) ∙ right-unit
+
+  left-left-inv : (p : x ＝ y) (q : y ＝ z) → inv p ∙ (p ∙ q) ＝ q
+  left-left-inv p q = inv (assoc (inv p) p q) ∙ ap (_∙ q) (left-inv p)
+```
```
Comment (EgbertRijke) [inline src/foundation/path-algebra.lagda.md:149]: These are duplications

https://unimath.github.io/agda-unimath/foundation-core.identity-types.html#concatenating-with-an-inverse-is-inverse-to-concatenating

**Review by EgbertRijke — APPROVED** [review]
This PR is pretty much flawless apart from one small duplication. Well done!

File: src/foundation/path-algebra.lagda.md (hunk 1 above)
Comment (fredrik-bakke) [reply src/foundation/path-algebra.lagda.md]: SMH is this like the 4th time I've added these lemmas due to forgetting the alternative interpretation? 🤦‍♂️ Either way, I've removed them again.

File: src/foundation/path-cosplit-maps.lagda.md [hunk 2]
```diff
@@ -151,6 +219,331 @@ is-trunc-map-is-path-cosplit-is-trunc-codomain k is-trunc-B is-cosplit-f =
...
+    is-path-cosplit-succ-coproduct F G N (inr x) (inl y) =
+      is-path-cosplit-is-equiv k
+        ( is-equiv-is-empty (ap (rec-coproduct f g)) (N y x ∘ inv))
+    is-path-cosplit-succ-coproduct F G N (inr x) (inr y) =
+      is-path-cosplit-equiv-arrow'
+        ( equiv-ap-emb emb-inr , id-equiv , λ where refl → refl)
+        ( G x y)
+```
+
+### Coroducts of path-cosplit maps are path-cosplit
```
Comment (fredrik-bakke) [inline src/foundation/path-cosplit-maps.lagda.md:500]:
```suggestion
### Coproducts of path-cosplit maps are path-cosplit
```

**Pull request merged**